src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -417,8 +415,10 @@ private static boolean shouldBeAloneOnLineWithNotAloneOption(Details details,
     private static boolean isAloneOnLine(Details details, String targetSrcLine) {
         final DetailAST rcurly = details.rcurly;
         final DetailAST nextToken = details.nextToken;
-        return (!TokenUtil.areOnSameLine(rcurly, nextToken) || skipDoubleBraceInstInit(details))
-                && CommonUtil.hasWhitespaceBefore(details.rcurly.getColumnNo(), targetSrcLine);
+        return (nextToken == null || !TokenUtil.areOnSameLine(rcurly, nextToken)
```

Comment:
So the way this reads, a `null` next token is always "alone on the line".

Can you show us a small example where one of these `null`s comes in?

Does a `null` next token always end up with no violations?

I ask to see if someone has checked if it makes more sense we add a global next token can't be `null` to the `validate` method  at https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java#L325 instead of all these little areas.

Reply:
> So the way this reads, a null next token is always "alone on the line".

Sir, what `null` only tells us that there is no token after it means it is the last of `file`, now the next part to check if there is there is something before it is handled by `CommonUtil.hasWhitespaceBefore(details.rcurly.getColumnNo(), targetSrcLine)`
and both are associated with and `&&` statement so both needs to be true. In short, no , if nextToken is `null` then it doesn't mean that token is alone.

> Can you show us a small example where one of these nulls comes in?


`public class InputRightCurlyEmptyOnSingleLine {} // violation`
-------------------------------------------------------------------------^----------

For this token, `nextToken` is `null`, since no next token is there, but it is not alone since `hasWhitespaceBefore()` is not statisfies.

> I ask to see if someone has checked if it makes more sense we add a global next token can't be null to the validate method at https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java#L325 instead of all these little areas.

I guess it would be tricky as well since we need to count multiple scenarios of `nextToken` being null. Also, future changes being easy for specific condition with this setup :)

Reply:
Ok, from all the NPEs that were gotten here in regression, I thought there was more going on then a simple class file's end right curly.